### DISCUSSION TITLE
Getting 500 error response with blank page on Vercel with no logs, but works fine in dev locally

### DISCUSSION THREAD

**Discussion by GabenGar** [vercel/next.js#37234] [Help] [answered]
Where to start digging?
The page in question is a file uploading form which gets processed on the server using [`formidable`](https://github.com/node-formidable/formidable), saves to the local temp folder, then another function reads said file and uploads it to the cloud storage. 
The process is a bit convoluted but mostly because `formidable` requires writing a ton of streaming boilerplate to get the `Buffer` from the form directly and I've read the lambdas do allow some form of read/write on their temporary fs and this whole process takes less than 10 seconds so it should work.
I assume by the nature of error Vercel blocks the request even before reaching the app hence the blank page instead of `/pages/500`, but how do I debug this?

**Comment by icyJoseph** [comment, marked as answer]
Is there any Vercel error you get to see?

> [Serverless Function Payload Size Limit](https://vercel.com/docs/concepts/limits/overview#serverless-function-payload-size-limit)
> The maximum payload size for the request body or the response body of a [Serverless Function](https://vercel.com/docs/concepts/functions/introduction) is 5 MB.
> If a Serverless Function receives a payload in excess of the limit it will return an error → 413: FUNCTION_PAYLOAD_TOO_LARGE.

I remember also wrestling with Formidable. I modified this quite a bit, but the idea is to skip writing to file system, and just keep it in memory, perhaps this helps?

```ts
import {Writable} from 'stream';

import formidable from 'formidable';
import {NextApiRequest, NextApiResponse} from 'next';

const formidableConfig = {
    keepExtensions: true,
    maxFileSize: 1_000_000,
    maxFieldsSize: 1_000_000,
    maxFields: 7,
    allowEmptyFiles: false,
    multiples: false,
};

function formidablePromise(
    req: NextApiRequest,
    opts?: Parameters<typeof formidable>[0]
): Promise<{fields: formidable.Fields; files: formidable.Files}> {
    return new Promise((accept, reject) => {
        const form = formidable(opts);

        form.parse(req, (err, fields, files) => {
            if (err) {
                return reject(err);
            }
            return accept({fields, files});
        });
    });
}

const fileConsumer = <T = unknown>(acc: T[]) => {
    const writable = new Writable({
        write: (chunk, _enc, next) => {
            acc.push(chunk);
            next();
        },
    });

    return writable;
};

export async function handler(req: NextApiRequest, res: NextApiResponse) {
    if (req.method !== 'POST') return res.status(404).end();

    try {
        const chunks: never[] = [];

        const {fields, files} = await formidablePromise(req, {
            ...formidableConfig,
            // consume this, otherwise formidable tries to save the file to disk
            fileWriteStreamHandler: () => fileConsumer(chunks),
        });

        const content = Buffer.concat(chunks);
        console.log({fields, files, content});

        return res.status(204).end();
    } catch (err) {
        return res.status(500).json({error: 'Internal Server Error'});
    }
}

export const config = {
    api: {
        bodyParser: false,
    },
};

export default handler;
```

**Reply by GabenGar** [reply]
That's the problem - there are no errors in the dashboard, not even a request log. Only 500 error returned by vercel. Forgot to mention the payload is below 5MB. But I'll try your solution.

**Reply by GabenGar** [reply]
I've tried your code and when I log the `content` value it shows a `Buffer` of zeroes, but `chunks` is in fact an array of `Buffer`s.
What gives?

**Comment by GabenGar** [comment]
Fixed it by changing `const content = Buffer.from(chunks);` to `const content = Buffer.concat(chunks);`

**Reply by icyJoseph** [reply]
Ah, when I wrote it here I made an error yeah, just checked my implementation. Updated answer.

**Reply by GabenGar** [reply]
Although now I get a timeout error. But at least it's an actual error in the logs.
These prod-only issues sure are annoying to debug.
